UPGRADING.md

```diff
@@ -1,6 +1,294 @@
+|   get_key(void, `InstanceHandle_t`*, bool) | compute_key(`SerializedPayload_t`&, `InstanceHandle_t`&, bool) |
+| DynamicPubSubType::createData() | DynamicPubSubType::create_data() |
+| DynamicPubSubType::deleteData(void*) | DynamicPubSubType::delete_data(void*) |
+| DynamicPubSubType::getKey(const void* const, `InstanceHand`*, bool) | DynamicPubSubType::compute_key(const void* const, `InstanceHand`&, bool) |
+| DynamicPubSubType::getSerializedSizeProvider(const void* const, `DataRepresentationId_t`) | DynamicPubSubType::calculate_serialized_size(const void* const, `DataRepresentationId_t`) |
+
+
+
+## Struct, Enum, Variable
+
+* Extend `SubscriptionBuiltinTopicData` with additional fields to mimic those of `ReaderProxyData`.
+* Extend `PublicationBuiltinTopicData` with additional fields to mimic those of `WriterProxyData`.
+* Extend `ParticipantBuiltinTopicData` adding the the Product version and additional fields to mimic those of `ParticipantProxyData`.
+* `DiscoveryProtocol_t` is `DiscoveryProtocol`.
+* Extend `SendBuffersAllocationAttributes` with a new attribute defining the allocation configuration of the `NetworkBuffers`.
+* `TypeConsistencyQos` was removed from DataReader, and `TypeConsistencyEnforcementQosPolicy` and `DataRepresentationQosPolicy` were included.
+* `initialHeartbeatDelay` is `initial_heartbeat_delay`.
+* `heartbeatPeriod` is `heartbeat_period`.
+* `nackResponseDelay` is `nack_response_delay`.
+* `nackSupressionDuration` is `nack_supression_duration`.
+* `heartbeatResponseDelay` is `heartbeat_response_delay`.
+* `initialAcknackDelay` is `initial_acknack_delay`.
+* `expectsInlineQos` is `expects_inline_qos`.
+* `m_typeSize` is `max_serialized_type_size`
+* `m_isGetKeyDefined` is `is_compute_key_provided`
+* `m_topicDataTypeName` is `topic_data_typename_`
+
+
+## Examples
+
+### Hello World
+Refactor the HelloWorld example with the current new example format.
+In this hello world example, the key changes are:
+* Removed --env option and set that environment behavior as default.
+* Added subscriber waitsets class and its usage.
+* Provided 	XML profiles examples that target several scenarios (e.g.,SampleConfig_Controller, Events, Multimedia).
+
+### X-Types Examples
+In this X-Types example, a type is defined at runtime on the publisher side using the Dynamic Types API, and the subscriber discovers the type, creates a reader for it, and prints the received data. This example is type compatible with the Hello World example, and a compatibility test has been added for this.
+
+### Configuration
+In this configuration example, the key changes are:
+* Included LargeData as an option (builtin transport argument).
+* Included all previous QoS examples:
+    * Deadline
+    * Disable positive ACKs
+    * Lifespan
+    * Liveliness
+    * Ownership (strength)
+    * Partitions
```

Comment:
Try to write this as a parragraph, as the suggestion in helloworld example. This applies also to the remain examples

Reply:
Don't you think it's clearer as a bullet list?